source/jsonParsing.mli

```diff
@@ -124,3 +124,71 @@ val optional_int_member : string -> Yojson.Safe.t -> int option
     - If [json] is indeed a directory but [name] is not presented, return [None].
     - In all other cases, raise {!Yojson.Safe.Util.Type_error}. *)
 val optional_path_member : string -> Yojson.Safe.t -> PyrePath.t option
+
+module Location : sig
+  type t = {
+    start_line: int;
+    end_line: int;
+    start_column: int;
+    end_column: int;
+  }
+  [@@deriving show, eq]
+
+  val from_decoded_range : (int * int) * (int * int) -> t
+end
+
+module Node : sig
+  type 'a t = {
+    location: Location.t option;
+    value: 'a;
+  }
+  [@@deriving show, eq]
+end
+
+exception
+  JsonmException of {
+    message: string;
+    location: Location.t;
+  }
+
+module Json : sig
+  type expression =
+    [ `Null
+    | `Bool of bool
+    | `String of string
+    | `Float of float
+    | `Int of int
+    | `List of t list
+    | `Assoc of (string * t) list
+    ]
+
+  and t = expression Node.t [@@deriving show, eq]
+
+  exception
+    TypeError of {
+      message: string;
+      json: t;
+    }
+
+  val json_of_string : string -> t
+
+  val null_node : t
+
+  module Util : sig
+    val keys : t -> string list
+
+    val member : string -> t -> t
+
+    val member_nullable : string -> t -> t
+
+    val to_bool_option : t -> bool option
+
+    val to_string : t -> string
+
+    val to_int : t -> int
+
+    val to_int_option : t -> int option
+
+    val to_list : t -> t list
+  end
+end
```

Comment:
I was intending to replace all instances of Yojson in favor of the new module, but yeah it might take some time and till then we can rename it to JsonAst and YojsonUtils :)